src/libraries/Common/src/Extensions/TypeNameHelper/TypeNameHelper.cs

```diff
@@ -49,9 +49,43 @@ internal static class TypeNameHelper
         /// <returns>The pretty printed type name.</returns>
         public static string GetTypeDisplayName(Type type, bool fullName = true, bool includeGenericParameterNames = false, bool includeGenericParameters = true, char nestedTypeDelimiter = DefaultNestedTypeDelimiter)
         {
-            var builder = new StringBuilder();
-            ProcessType(builder, type, new DisplayNameOptions(fullName, includeGenericParameterNames, includeGenericParameters, nestedTypeDelimiter));
-            return builder.ToString();
+            if (type.IsGenericType)
```

Comment:
Instead of duplicating all of this logic, could:
```C#
private static void ProcessType(StringBuilder builder, Type type, in DisplayNameOptions options)
```
be changed to instead be something like:
```C#
private static string? ProcessType(ref StringBuilder? builder, Type type, in DisplayNameOptions options)
```
where it would append to the builder if it's not null, lazily initialize the builder if it needs to append more than one thing, but if the builder is null and it only has one string then just return it?

Reply:
Updated